### PR TITLE
Add ActivationQuantizationHolder and use it during GPTQ

### PR DESCRIPTION
**Pull request by reuvenperetz**
Add ActivationQuantizationHolder layer to hold an activation quantizer and quantize during inference.
Update Keras model builder to use it as quantization layers if it is passed (instead of
KerasQuantizeWrapper that was used for both weights and activations).
For now, only GPTQ uses it by passing it to the model builder - thus, GPTQ models wrap layers
with weight quantizers and quantize activations using ActivationQuantizationHolder layers.

### CODE REVIEW

File: model_compression_toolkit/qat/keras/quantization_facade.py [hunk 1]
```diff
@@ -56,19 +56,27 @@
     DEFAULT_KERAS_TPC = get_target_platform_capabilities(TENSORFLOW, DEFAULT_TP_MODEL)
 
 
-    def qat_wrapper(n: common.BaseNode, layer: Layer, qat_config):
+    def qat_wrapper(n: common.BaseNode,
```
Comment (haihabi) [inline model_compression_toolkit/qat/keras/quantization_facade.py:59]: Why do we need to change QAT?

File: model_compression_toolkit/qat/keras/quantization_facade.py [hunk 2]
```diff
@@ -56,19 +56,27 @@
...
+            wrap_with_activation_quantizers: Whether to use the wrapper for the activation quantizer or not
+
 
         Returns: Wrapped layer
 
         """
         if _is_qat_applicable(n, DEFAULT_KERAS_INFO):
             weights_quantizers, activation_quantizers = quantization_builder(n, qat_config, DEFAULT_KERAS_INFO)
-            return qi.KerasQuantizationWrapper(layer, weights_quantizers, activation_quantizers)
+            if wrap_with_activation_quantizers:
```
Comment (ofirgo) [inline model_compression_toolkit/qat/keras/quantization_facade.py:76]: Update this according to what you decide regarding the terminology of "wrapper"/"holder".

File: tests/keras_tests/feature_networks_tests/feature_networks/gptq/gptq_test.py [hunk 3]
```diff
@@ -44,7 +44,8 @@ def build_model(in_input_shape: List[int]) -> keras.Model:
 
     """
     inputs = layers.Input(shape=in_input_shape)
-    x = layers.Conv2D(64, 4, bias_initializer='glorot_uniform')(inputs)
+    x = layers.Conv2D(64, 4)(inputs)
```
Comment (ofirgo) [inline tests/keras_tests/feature_networks_tests/feature_networks/gptq/gptq_test.py:47]: add function test to test graph builder

**Review by ofirgo — CHANGES_REQUESTED** [review]

File: model_compression_toolkit/core/keras/back2framework/keras_model_builder.py [hunk 4]
```diff
@@ -211,9 +226,13 @@ def build_model(self) -> Tuple[Model, UserInformation]:
             def _wrap(layer):
                 _node = self.oh.layer_to_node_dict.get(layer)
                 if _node is not None:
-                    return self.wrapper(_node, layer)
-                elif is_layer_fake_quant(layer):
+                    return self.wrapper(_node,
+                                        layer,
+                                        wrap_with_activation_quantizers=self.use_activation_quantizers_in_wrapper)
```
[the remark this answers is not in the document]
Comment (reuvenperetz) [reply model_compression_toolkit/core/keras/back2framework/keras_model_builder.py:231]: So it'll be possible to do it in phases in different PRs (next PRs: QAT, Exporter, etc.)

**Pull request merged**